src/base/api.lua

```diff
@@ -1092,6 +1092,31 @@
 	})
 
 
+--
+-- nested data kind; wraps simple values into a table, returns others as-is.
+--
+
+	premake.field.kind("nested", {
+		store = function(field, current, value, processor)
+			-- set each value.
+			local result = {}
+			for name, v in pairs(value) do
+				local f = field.fields[name]
```

Comment:
I could be wrong, but I think this might cause a problem if the global API has a field with the same name, or even a different nested field. I think the name might need to be decorated like `schemacompiler.template` to avoid collisions.